scripts/update_supported_distros.py

```diff
@@ -0,0 +1,199 @@
+            if release['isEol'] is False:
+                supported.append(f"{distro}-{release['name']}")
+
+    return supported
+
+
+def get_current_distros():
+    try:
+        current = os.path.dirname(os.path.abspath(__file__))
+        path = os.path.join(current, "..", ".github", "workflows", "distro.yml")
+        
+        with open(path, 'r') as f:
+            data = yaml.safe_load(f)
+        
+        matrix = data['jobs']['distro-tests']['strategy']['matrix']['include']
+        distros = [d for d in matrix if d.get('distro') not in IGNORED_DISTROS]
+
+        return distros
+    
+    except Exception as e:
+        print(f"Error reading distro.yml file: {e}")
+        return None
+
+
+def remove_distro(distro):
+    current = os.path.dirname(os.path.abspath(__file__))
+    path = os.path.join(current, "..", ".github", "workflows", "distro.yml")
+    
+    yaml_parser = YAML()
+    yaml_parser.preserve_quotes = True
+    yaml_parser.width = 4096  
+    yaml_parser.indent(mapping=2, sequence=4, offset=2)
+    
+    with open(path, 'r') as f:
+        data = yaml_parser.load(f)
+    
+    matrix = data['jobs']['distro-tests']['strategy']['matrix']['include']
+    
+    for i, entry in enumerate(matrix):
+        if entry.get('distro') == distro:
+            matrix.pop(i)
+            break
+    
+    # Write the new matrix back to the distro.yml file
+    with open(path, 'w') as f:
+        yaml_parser.dump(data, f)
+    
+    # Post-process to fix yaml_parser formatting (Replaces |- with | for run fields)
+    with open(path, 'r') as f:
+        content = f.read()
+
+    content = re.sub(r'(\s+run:\s*)(\|-)(\s*\n)', r'\1|\3', content)
+    
+    with open(path, 'w') as f:
+        f.write(content)
+        
+    
+def add_distro(distro):
+    current = os.path.dirname(os.path.abspath(__file__))
+    path = os.path.join(current, "..", ".github", "workflows", "distro.yml")
+    
+    parts = distro.split('-')
+    distro_name = parts[0]
+    distro_version = '-'.join(parts[1:])
+    
+    template = DISTRO_TEMPLATES[distro_name]
+    
+    yaml_parser = YAML()
+    yaml_parser.preserve_quotes = True
+    yaml_parser.width = 4096  
+    yaml_parser.indent(mapping=2, sequence=4, offset=2)
+    
+    with open(path, 'r') as f:
+        data = yaml_parser.load(f)
+    
+    matrix = data['jobs']['distro-tests']['strategy']['matrix']['include']
+    
+    # Find where to insert the new distro (after the last occurrence of the same distro type). If none found, append to the end.
+    insert_index = len(matrix)
+    for i, entry in enumerate(matrix):
+        if entry.get('distro', '').startswith(distro_name + '-'):
+            # Find the last occurrence of this distro type
+            for j in range(i + 1, len(matrix)):
+                if not matrix[j].get('distro', '').startswith(distro_name + '-'):
+                    insert_index = j
+                    break
+            else:
+                insert_index = len(matrix)
+    
```

Comment:
This is extremely buggy code without tests. You could have Yaml to dict and update keys in dict then convert back and have way safer way to update data.